shared/libraries/websocket_streaming/src/streaming_client.cpp

```diff
@@ -235,92 +245,149 @@ void StreamingClient::onMessage(const daq::streaming_protocol::SubscribedSignal&
+                                 {
+                                     return tableId == pair.second->getTableId();
+                                 });
+    if (signalIt != signals.end())
     {
-        auto domainDescriptor = inputSignal->getDomainSignalDescriptor();
-        onDomainDescriptorCallback(tableId, domainDescriptor);
-
-        if (auto iterator = signalInitializedStatus.find(tableId); iterator != signalInitializedStatus.end())
-        {
-            iterator->second.first.set_value();
-        }
+        // value signal with tableId is known, set domain descriptor for it
+        setDomainDescriptor(signalIt->first, signalIt->second, sInfo.dataDescriptor);
+    }
+    else
+    {
+        // value signal with tableId is unknown, save domain descriptor
+        cachedDomainDescriptors.insert_or_assign(tableId, sInfo.dataDescriptor);
     }
 }
 
-void StreamingClient::publishSignal(const std::string& signalId)
+void StreamingClient::publishSignal(const SubscribedSignal& subscribedSignal)
 {
-    if (signals.count(signalId) == 0)
-        return;
-
-    auto inputSignal = signals[signalId];
-    if (!inputSignal->hasDescriptors())
-        return;
+    std::string tableId = subscribedSignal.tableId();
+    auto signalIt = std::find_if(signals.begin(),
+                                 signals.end(),
+                                 [tableId](const std::pair<std::string, InputSignalPtr>& pair)
+                                 {
+                                     return tableId == pair.second->getTableId();
+                                 });
+    if (signalIt != signals.end())
+    {
+        auto inputSignal = signalIt->second;
+        auto signalId = signalIt->first;
 
-    auto eventPacket = inputSignal->createDecriptorChangedPacket();
-    onPacketCallback(signalId, eventPacket);
+        // signal meta information is always received by pairs of META_METHOD_SIGNAL:
+        // one is meta for data signal, another is meta for time signal.
+        // we generate event packet only after both meta are received
+        // and all signal descriptors are assigned.
+        if (!inputSignal->hasDescriptors())
+            return;
+
+        setSignalInitSatisfied(signalId);
+        auto eventPacket = inputSignal->createDecriptorChangedPacket();
+        onPacketCallback(signalId, eventPacket);
+    }
 }
 
 void StreamingClient::onSignal(const daq::streaming_protocol::SubscribedSignal& subscribedSignal, const nlohmann::json& params)
 {
     try
     {
+        {
+            LOG_I("Signal #{}; signalId {}; tableId {}; name {}; value type {}; Json parameters: \n\n{}\n",
+                  subscribedSignal.signalNumber(),
+                  subscribedSignal.signalId(),
+                  subscribedSignal.tableId(),
+                  subscribedSignal.memberName(),
+                  subscribedSignal.dataValueType(),
+                  params.dump());
```

Comment:
Should these be info or debug level?

Reply:
I'd prefer to keep this at the 'info' level to have this logs in release versions